pkg/test/utils/kubernetes.go

```diff
@@ -870,7 +878,9 @@ func WaitForDelete(c *RetryClient, objs []runtime.Object) error {
 	// Wait for resources to be deleted.
 	return wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (done bool, err error) {
 		for _, obj := range objs {
-			err = c.Get(context.TODO(), ObjectKey(obj), obj.DeepCopyObject())
+			actual := &unstructured.Unstructured{}
+			actual.SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
+			err = c.Get(context.TODO(), ObjectKey(obj), actual)
```

Comment:
I've seen this pattern a couple times in the new code, maybe there's a way to unify it? It's not too bad though, I'd be ok with leaving it as is